service/src/main/java/bio/terra/workspace/service/workspace/WorkspaceService.java

```diff
@@ -285,4 +296,52 @@ public String getRequiredGcpProject(UUID workspaceId) {
         .getGcpCloudContext()
         .map(GcpCloudContext::getGcpProjectId);
   }
+
+  /**
+   * Grant a user permission to impersonate their pet service account in a given workspace. Unlike
+   * other operations, this does not run in a flight because it only requires one write operation.
+   * This operation is idempotent.
+   *
+   * @return The email identifier of the user's pet SA in the given workspace.
+   */
+  public String enablePet(UUID workspaceId, AuthenticatedUserRequest userRequest) {
+    final String serviceAccountUserRole = "roles/iam.serviceAccountUser";
+    Workspace workspace =
+        validateWorkspaceAndAction(
+            userRequest, workspaceId, SamConstants.SAM_WORKSPACE_WRITE_ACTION);
```

Comment:
Why only writers? Can readers not run workflows on data in the workspace?
(Maybe they can't. I don't know.)

Reply:
Yep, the current permissions model is that readers cannot run workflows, only writers + owners. Added a comment to clarify